src/check-image.py

```
@@ -1,14 +1,30 @@
+import warnings
+# Suppress urllib3 OpenSSL warning on macOS with LibreSSL before importing requests
+warnings.filterwarnings('ignore', message='urllib3 v2 only supports OpenSSL')
+
 import requests
 import sys
 import os
 import argparse
-from typing import List, Dict, Tuple
+from typing import List, Dict, Tuple, Optional
 
 # Target architectures to check
 TARGET_ARCHITECTURES = {'amd64', 'arm64'}
 TIMEOUT_SECONDS = 10
 
-def get_auth_token(repository: str) -> str:
+def detect_registry(image: str) -> str:
+    """Detect the registry type from the image name."""
+    if image.startswith('ghcr.io/'):
+        return 'ghcr'
+    elif image.startswith('quay.io/'):
+        return 'quay'
+    elif image.startswith('docker.io/') or '/' not in image or image.count('/') == 1:
+        return 'dockerhub'
+    else:
+        # Default to dockerhub for backward compatibility
+        return 'dockerhub'
```

Looks like this could ultimately give data about a completely different container if people input images that are in their own public registries (i.e. myregistry.com/image_name would eventually just check the docker image_name, which might be a different image to myregistry.com/image_name. It would be super misleading if we gave info about a completely different image.

---

Thanks Joe, added a fix in https://github.com/arm/arm-linux-migration-tools/pull/11/commits/c0d450fd1f3db53b9af1d0aa523b1431e4be312d to avoid silently defaulting to DockerHub for the image.